mcs/class/corlib/corlib-net_4_x.csproj

```diff
@@ -96,8 +96,6 @@
     <Compile Include="..\..\..\external\corefx\src\System.Security.Cryptography.X509Certificates\src\System\Security\Cryptography\X509Certificates\X509KeyStorageFlags.cs" />
     <Compile Include="..\..\..\external\corefx\src\System.Threading.Tasks.Extensions\src\System\Runtime\CompilerServices\AsyncMethodBuilderAttribute.cs" />
     <Compile Include="..\..\..\external\corefx\src\System.Threading.Tasks.Extensions\src\System\Runtime\CompilerServices\AsyncValueTaskMethodBuilder.cs" />
-    <Compile Include="..\..\..\external\corefx\src\System.Threading.Tasks.Extensions\src\System\Runtime\CompilerServices\ConfiguredValueTaskAwaitable.cs" />

```

Comment:
Two notes:
1. If you're editing the .csproj files, you really should be editing the `corlib.dll.sources` file.
2. This seems like a step back - we should aim to share more code, not to make our own copies.  If there is something that only makes sense for Mono it should be placed under an `#ifdef` and upstreamed.

Reply:
They are from coreclr rather than corefx and [the mono fork](https://github.com/mono/coreclr) is 13058 commits behind dotnet:master.

How should I proceed?

Reply:
For these types the corefx versions are type-forwarded to the coreclr versions

Reply:
For these types the corefx versions (NuGet) are type-forwarded to the coreclr (runtime) versions; might be why they aren't actually in  .sources files (e.g. NuGet sourced)

Reply:
I'm really confused about corefx - we should probably wait for someone who knows what's going on.  Maybe @akoeplinger  can help.

There seem to be two copies of `ConfiguredValueTaskAwaitable.cs` in corefx

1. One in `System.Threading.Tasks.Extensions` https://github.com/dotnet/corefx/blob/45b724f6b6391910edea8a70f3f22a4a7996696d/src/System.Threading.Tasks.Extensions/src/System/Runtime/CompilerServices/ConfiguredValueTaskAwaitable.cs
2. One in `CoreLib` https://github.com/dotnet/corefx/blob/103639b6ff5aa6ab6097f70732530e411817f09b/src/Common/src/CoreLib/System/Runtime/CompilerServices/ConfiguredValueTaskAwaitable.cs

The `CoreLib` `ConfiguredValueTaskAwaitable.cs` is identical to the one in your PR.  So we should just include the `CoreLib` one from corefx.

Reply:
The 90% case is covered by not changing these two file (as `ValueTask<TResult>` isn't widely used atm); so will back out those changes

Reply:
These files are generated automatically, you should modify `mcs/class/corlib/corlib.dll.sources` directly. Simply add the following right after https://github.com/mono/mono/blob/master/mcs/class/corlib/corlib.dll.sources#L1623

```
coreclr/StackTraceHiddenAttribute.cs
coreclr/ConfiguredValueTaskAwaitable.cs
coreclr/ValueTaskAwaiter.cs
```

Reply:
@lambdageek @benaadams we don't use the coreclr fork in the Mono build, it's mainly used to grab some tests which is why we don't regularly update it. We only use corefx and corert.

Reply:
I see now that `corlib.dll.sources` supports wildcards.

```
../../../external/corefx/src/System.Threading.Tasks.Extensions/src/System/Runtime/CompilerServices/*.cs
```
We should probably not do that and get `src/Common/src/CoreLib/System/Runtime/CompilerServices/ConfiguredValueTaskAwaitable.cs` instead.